### ISSUE TITLE
Unexpected move following alrm reset

### ISSUE THREAD

**Issue by micksulley** [vlachoudis/bCNC#1216]
Running 0.9.15 on a Raspberry Pi with an Ox
I have seen this a couple of times now, this time I was sure what happened.
I did a probe to set Z, jogged Z up a bit then jogged X & Y to my work zero, jogged Z down again to Z zero.
I then told it to probe and I got an error, which is expected as it is already sensing zero point.
I then pressed Reset, then Unlock.
At this point it set off at max speed in both +X and +Y, Z did not move.  Hit the emergency stop!
Why did it move at all?  I had not seen this on my previous version
Thanks
Mick

**Comment by micksulley** [comment]
I have done more testing and this is reproducible. 
Probe with Z already triggered (not a normal occurrence but can happen)
Alarm(Z)  (as expected)
Reset
Unlock
Off goes the carriage in both X and Y.  My view is that is should need a positive action,e.g. start or resume, before anything moves at all.

**Comment by Harvie** [comment]
Can this be related to #1208 ?

**Comment by micksulley** [comment]
I have looked at #1208 and whilst it may be related I don't believe that it is the same thing.  The problem I see is that the machine moves on an alarm reset/unlock.  I haven't told it to do anything at all, just reset/unlock and off it goes.  As described above it is repeatable for me, does it behave this way for you?  I am happy to do further tests if there is any other information you need.

**Comment by Harvie** [comment]
Do you use GRBL? Can you send us your $$ config?

**Comment by micksulley** [comment]
Yes it is GRBL
$0=10
$1=255
$2=0
$3=6
$4=1
$5=0
$6=0
$10=1
$11=0.010
$12=0.002
$13=0
$20=0
$21=1
$22=1
$23=3
$24=25.000
$25=2500.000
$26=250
$27=1.000
$30=1000
$31=0
$32=0
$100=26.662
$101=26.641
$102=200.000
$110=2500.000
$111=2500.000
$112=750.000
$120=50.000
$121=50.000
$122=50.000
$130=820.000
$131=775.000
$132=68.000
Let me know if you need any more info or want me to try anything.
Mick

**Comment by Harvie** [comment]
`$22=1` means you have homing cycle enabled...

**Comment by MARIOBASZ** [comment]
I use homing cycle, but I also use soft limits $ 20 = 1, if I mistake the distance outside the limits, it avoids shocks. I have not suffered this problem.

**Comment by MARIOBASZ** [comment]
Start cycle is very useful if I have to restart at the same point

**Comment by micksulley** [comment]
When I enter $N I see -
$N
$N0=
$N1=
ok
$G
[GC:G0 G54 G17 G21 G90 G94 M5 M9 To F0 S0]
ok

I tried setting $22=0  The problem is still there just the same, also the Home button does not work (obvious I suppose) so this is really not an option for me.
I also looked at the terminal output as the problem occurs, it shows
G53 G0 X0 Y0 Z0
I believe that G53 is absolute coordinates, so it is not driving to home but to absolute zero.

Why does it drive anywhere following a reset/unlock?  Surely you should have to give it an instruction before it moves at all?
If there is some reason to auto home it following a reset/unlock then 1) it should be to machine home not absolute zero and 2) it should home Z first to get the tool out of the way and only then home X and Y

**Comment by Harvie** [comment]
> I tried setting $22=0 The problem is still there just the same.

Well so it's probably some other problem... Can you please launch bCNC from commandline and use `spy://` option in serial connection menu to capture serial communication?

**Comment by micksulley** [comment]
Yes if you tell me how to use the spy:// option

**Comment by Harvie** [comment]
![image](https://user-images.githubusercontent.com/276504/57291738-4da1ec80-70c0-11e9-888d-61321db1d704.png)

**Comment by micksulley** [comment]
I didn't realise that option existed :)
I have taken some screenshot which hopefully give the necessary data.
Screenshot2 is the point of the alarm, 3 is the reset/unlock 4 & 5 show the GRBL version (I think)

![Screenshot-2](https://user-images.githubusercontent.com/6191767/57298295-c194c280-70c9-11e9-8762-c60d24fef6bf.png)
![Screenshot-3](https://user-images.githubusercontent.com/6191767/57298296-c194c280-70c9-11e9-83c3-a06a2bb75194.png)
![Screenshot-4](https://user-images.githubusercontent.com/6191767/57298297-c194c280-70c9-11e9-8083-5deebd2ffdf2.png)
![Screenshot-5](https://user-images.githubusercontent.com/6191767/57298298-c22d5900-70c9-11e9-832c-da784cfa5bea.png)

**Comment by Harvie** [comment]
This seems to be the problem, but i don't know what causes it:

![image](https://user-images.githubusercontent.com/276504/57302126-c3b34d00-70da-11e9-9e66-7997f0941f93.png)

What do you think?

**Comment by micksulley** [comment]
Yes looks like it.  How/why is that instruction generated?  This problem certainly didn't happen on the previous version that I ran.

**Comment by micksulley** [comment]
Also I believe that this problem only occurs on reset/unlock following a probe Z already triggered alarm, not seen it after any other reset/unlock condition.

**Comment by Harvie** [comment]
According to grep, there's only one line which can send such command:

```
./ProbePage.py:			cmd = "G53 G0 X%g Y%g Z%g\n"%(CNC.vars["prbx"], CNC.vars["prby"], CNC.vars["prbz"])
```

Maybe it gets buffered somewhere on bCNC side and the buffer is not cleared during reset, so it gets sent afterwards... We should clean that buffer during reset...

**Comment by Harvie** [comment]
> That line is exactly the same in the version I was running before, did't have the problem with that

I know, but new version might buffer this line in different manner...

**Comment by GitHubCaps** [comment]
@micksulley could this possibly be related to #1389 ? See the reply about "Automatic GOTO after probing" check-box. I realize this is an old post, but if others are having similar issues as this, there may be a need to handle this differently to achieve the desired results.

**Comment by micksulley** [comment]
I have updated both issues, it does look to be very similar if not the 
same issue.

Mick
